### DISCUSSION TITLE
v3 acceptance test failures with FileNotFoundException

### DISCUSSION THREAD

**Discussion by Xcodo** [xunit/xunit#2328] [Question] [answered]
I was attempting to work on a fix for #2232 but after overriding the equality comparisons in `_NullMessageSink` the acceptance tests start failing somewhat erratically on my local machine with `FileNotFoundExceptions` referring to being unable to load an assembly, for example
```
Xunit3TheoryAcceptanceTests+CustomDataTests.TestDataWithInternalConstructor_ReturnsTwoPassingTheories [FAIL]
      System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation.
      ---- System.IO.FileNotFoundException : Could not load file or assembly 'assembly-cc717c6cbf2f4d98819f7316c2f9c791, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
```

See https://github.com/Xcodo/xunit/runs/2935468991?check_suite_focus=true#step:8:188 for an example of it failing.

Any idea why this is happening? The lack of repeatability or debugging when running with the build script means I'm a bit at a loss.

**Comment by bradwilson** [comment, marked as answer]
I've definitely seen this before but I don't recall what the cause was. Are you on the latest code from `main`?

**Reply by Xcodo** [reply]
I was last week when I had the issue. Is there support anywhere for running v3 in a debugger yet? If there is I can have a better poke around when the issue occurs and see what DLL artifacts have been generated when the failure occurs.

**Reply by bradwilson** [reply]
We don't yet have support for the Visual Studio Test Explorer, because we haven't finished v3 support in Runner Utility. That means Test Explorer is not usable yet.

That said, v3 test projects are programs, so they can be debugged just like programs. Set a breakpoint where you want it, then choose the test project to debug from the drop down in the Standard toolbar, ala:

![image](https://user-images.githubusercontent.com/16944/125121134-4fa0d500-e0a8-11eb-84d1-ba789d759606.png)

Then you should be able to debug the test assembly.

You can also speed up debugging by choosing just a single test class to run via the debug command line options in the project settings, like this:

![image](https://user-images.githubusercontent.com/16944/125121416-bfaf5b00-e0a8-11eb-8e0e-96baa341fd40.png)

You can even specify a single method to run, like this:

![image](https://user-images.githubusercontent.com/16944/125121510-e372a100-e0a8-11eb-9783-7e47971ce597.png)

You can combine multiple `-class` or `-method` command line options if you need to run just a small set of test classes or test methods.
